src/app/home/home.component.ts

```diff
@@ -42,7 +75,20 @@ export class HomeComponent implements OnInit, OnDestroy {
     );
   }
 
-  ngOnDestroy() {
-    this.dataSubscription$.unsubscribe();
+  onLoadChartData(type: string) {
+    this.loadChartData(type);
+  }
+
+  loadChartData(type: string) {
+    this.activeSort = joinString(type);
+
+    this.tableData = this.chartData
+      .sort((a: ICountryInfo, b: ICountryInfo) =>
+        a[this.activeSort] < b[this.activeSort] ? 1 : -1
+      )
+      .slice(0, this.numberOfData);
+    this.lineChartData = chartDataLoader(this.lineChartData, this.tableData);
+    this.lineChartLabels = this.tableData.map((el) => el.Country);
+    this.text = `${this.numberOfData} countries with most ${this.activeSort} cases`;
```

Comment:
Since you are separating `activeSort` and `tableData` with an empty row, feel free to do so with other parts, as well.

e.g.

activeSort / tableData / linChartData & lineChartLabels / text